### ISSUE TITLE
Support for index signatures

### ISSUE THREAD

**Issue by manast-apsis** [lukeautry/tsoa#374]
Seems like index signatures such as
```
class myClass {
    nodes: {
        [nodeName: string]: Node;
    };
}
```
are not currently supported.

https://www.typescriptlang.org/docs/handbook/advanced-types.html#index-types-and-string-index-signatures

nodes will just show as an empty object in the swagger file.

**Comment by dgreene1** [comment]
I'll look into making a PR for this after https://github.com/lukeautry/tsoa/pull/385 merges (since that's a big PR with lots of changes.

In the meantime, although the swagger output will not contain the correct information, the request validation will ensure that the data is correct. There is an API test that confirms that the following interface actually gets validated.
```ts
export interface ValidateMapStringToNumber {
  [key: string]: number;
}
```
And here's the unit test that proves the validation works as expected: https://github.com/lukeautry/tsoa/blob/a5ee2904ae12ddb9d3951f85304be5f26f4560c9/tests/integration/express-server.spec.ts#L479

So long story short, please continue to use `tsoa` for your validation, and we'll add a PR for better swagger generation for indexed types soon. :)

I know it's a bit strange that tsoa's swagger output is separate from it's runtime validation, but that's advantageous because we return nice, human readable output.

**Comment by javichi** [comment]
Looking forward for this feature :)

**Comment by dgreene1** [comment]
I'm closing this because it's a duplicate of #50. However, until #50 is solved, the workaround for this particular case is:
```ts
interface INodeDictionary {
    [nodeName: string]: Node;
}
class myClass {
    nodes: INodeDictionary;
}
```

**Comment by dgreene1** [comment]
We need to reopen this to make sure that the case presented by @javichi works: https://github.com/lukeautry/tsoa/issues/50#issuecomment-528213658

**Comment by dgreene1** [comment]
@javichi / @manast-apsis this fix/feature was just [released in v2.5.1](https://github.com/lukeautry/tsoa/releases). Can you let us know how it worked out for you?

Side note: we're looking for companies that want to be featured in tsoa's readme: https://github.com/lukeautry/tsoa/issues/464

**Comment by javichi** [comment]
Sorry for the delay. I can confirm that all index signatures are generated in v2.5.1. Great work guys!
